src/high-level/allocation.lisp

```diff
@@ -20,14 +20,34 @@
     1. The allocated storage.
     2. A finalizer thunk of type FINALIZER, which should be called when the memory is OK to be freed.
 NOTE: Note that the finalizer may close over the allocated vector."
-  '(function (integer &key (:element-type t) (:initial-element t)) (values (simple-array *) finalizer)))
+  '(function (integer t t) (values (simple-array *) finalizer)))
 
 
 ;;;;;;;;;;;;;;;;;;;;;;;; Lisp Heap Allocation ;;;;;;;;;;;;;;;;;;;;;;;;
 
 (declaim (ftype allocator lisp-allocator))
-(defun lisp-allocator (size &key element-type initial-element)
+(defun lisp-allocator (size element-type initial-element)
   (let ((storage
+          #+allegro ;; low-level optimizations (inlining) for Allegro
```

Comment:
In regard to the performance, Allegro currently couldn't inline `make-array` because `initial-element` can be either `nil` (which yields an uninitialized array) or a number. We are trying to implement an optimisation here so that `lisp-allocator` can be much simplified here.

My suggestion would be to create a new issue. When our optimisation is ready, I'll revise it in the future. Does it make sense to you?

Reply:
Can we put this chunk of code in a separate file in the ASD and `:if-feature :allegro` it? I understand why it needs to be here but I would prefer to still separate it.